src/org/way2it/data_structures/list/List.java

```diff
@@ -14,30 +16,78 @@ public List() {
     // Should add new value to the end of the list, increment length and extend array by x2 if needed
     public void add(String value) {
         // TODO implement me
+        valuesInc();
+        values[length]=value;
+        length++;
     }
 
     // Should add new value at the specified index, moving other values to the right,
     // increment length and extend array by x2 if needed
     public void addAtIndex(String value, int index) {
         // TODO implement me
+        valuesInc();
+
+        String[] temp = values;
+        values = new String[temp.length];
+        int k = 0;
+        for (int i = 0; i<temp.length;i++){
+            if (i == index){
+                values[i]=value;
+                i++;
+            }
+            values[i]=temp[k];
+            k++;
+        }
+
+        length++;
+    }
+
+    private void valuesInc() {
+        if (length == values.length) {
+            String[] temp = values;
+            int newLength = length * 2;
+            values = new String[newLength];
+            System.arraycopy(temp, 0, values, 0, temp.length);
+        }
     }
 
     // Should return a value at specified index, or null if index is out of list bounds
     public String get(int index) {
         // TODO implement me
-        return null;
+      if(index < values.length){
+          return values[index];
+      }
+        else return null;
     }
 
     // Should return the index of the first occurrence of specified value in list
     // If the value does not exist - return -1
     public int indexOf(String value) {
         // TODO implement me
-        return -1;
+        int k=-1;
+        for(int i = 0; i < values.length;i++){
+            if (Objects.equals(values[i], value)){
+               k=i;
```

Comment:
As you need to find exactly the first occurrence of the value, you should place a break here.